### PR TITLE
[FISH-501] - Update for Jakarta EE9

### PR DESCRIPTION
**Pull request by cubastanley**
Includes updates for running the Jakarta EE9 CDI TCK - important to note that this doesn't result in passing tests, just tests that run, passing tests comes later

### CODE REVIEW

File: build.properties [hunk 1]
```diff
@@ -51,5 +51,5 @@ max.perm.gen.size=1024m
 # found under ${porting.home}/glassfish-tck-runner/target/surefire-reports
 ######################################################
 report.dir=${porting.home}/reports
-cdiextjar=cdi-tck-ext-lib-2.0.6.jar
-cdiext.version=2.0.6
+cdiextjar=cdi-tck-ext-lib-3.0.0.jar
```
Comment (MarkWareham) [inline build.properties:54]: this will need hosting

File: settings.xml [hunk 2]
```diff
@@ -1,54 +1,62 @@
...
     Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
     version 2 with the GNU Classpath Exception, which is available at
     https://www.gnu.org/software/classpath/license.html.
-
     SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
-
 -->
 
 <settings>
+      <localRepository>${env.PORTING}/.m2/repository</localRepository>
```
Comment (MarkWareham) [inline settings.xml]: This is going to slow things down. This dir on CI will be created new each time, so all dependencies will be needed to be downloaded for each module!

File: settings.xml [hunk 3]
```diff
@@ -1,54 +1,62 @@
...
       <snapshots>
         <enabled>true</enabled>
       </snapshots>
-    </pluginRepository>
-  </pluginRepositories>
+    </repository>
+    <repository>
+      <id>jakarta</id>
+      <name>Jakarta Staging Repository</name>
+      <url>https://jakarta.oss.sonatype.org/content/repositories/staging/</url>
```
Comment (MarkWareham) [inline settings.xml]: What's this? We probably don't want to use any artifacts from here

File: build.xml [hunk 4]
```diff
@@ -522,8 +530,8 @@
 			<arg value="/c"/>
                         <arg value="${maven.executable}.cmd"/>
                         <arg value="dependency:get"/>
-						<arg value="-DrepositoryUrl=https://repo1.maven.org/maven2/"/>
-						<arg value="-DgroupId=org.jboss.cdi.tck"/>
+						<arg value="-DrepositoryUrl=https://jakarta.oss.sonatype.org/content/repositories/staging"/>
```
Comment (MarkWareham) [inline build.xml]: Is there o release repo? - do we want to get from staging repo?

File: glassfish-tck-runner/pom.template.xml [hunk 5]
```diff
@@ -59,16 +59,23 @@
             <dependency>
                 <groupId>jakarta.xml.ws</groupId>
                 <artifactId>jakarta.xml.ws-api</artifactId>
-                <version>2.3.2</version> <!-- 2.3.3 is not in central yet -->
+                <version>3.0.0-RC3</version> <!-- 2.3.3 is not in central yet -->
```
Comment (MarkWareham) [inline glassfish-tck-runner/pom.template.xml]: comment of no use

**Review by MarkWareham — APPROVED** [review]
Left a few comments, some that will need thought and may or may not require actions. 

Seem to be using RCs and staging repos a lot. Assuming this is because still a WIP with eclipse, will need to ensure this will get updated when the final artifacts are ready.

**Comment by MarkWareham** [comment]
>Use releases repo...
Can any of the RCs being used be made into final artifact versions?

**Comment by cubastanley** [comment]
> > Use releases repo...
> > Can any of the RCs being used be made into final artifact versions?

Unfortunately not, no - it's still RCs in the releases repo in the case that they're there (although there are only two which isn't too many in the sea of dependencies required) - It'll be tasked to update them on full release of the EE9 artifacts

**Pull request merged**